### ISSUE TITLE
Gallery needs to be fixed

### ISSUE THREAD

**Issue by holger1411** [understrap/understrap#115]
See discussions here:
https://github.com/holger1411/understrap/pull/75

And was reported here:
https://github.com/holger1411/understrap/issues/74

The Problem (By described @flawiddsouza):

> Using the current implementation, some of the images included in the gallery shortcode snippet just go missing and are not included for some reason. Though, I'm unaware of the real intention for this pull request. I'm using it in production and all's good but it might not be what everyone wants.

**Comment by Thomas-A-Reinert** [comment]
Hm, I can´t reproduce why some images don´t seem to show up for @flawiddsouza . They ALL do on my local install. But I CAN confirm that they all show up one below the other. They don´t get the col-xx-x attached for some reason.

And I can also confirm that the roots.io way works perfectly. There are still some mods to it that would make perfect sense, but *HEY* it works though. We could enhance that further once its included.

As roots.io´s solution is GPL`ed too as far as I know it shouldn´t be a problem to include it into this project as long as a reference to it remains within the code. Anyway, I would get in contact with them and double-check, if I was you, Holger. So I´ll not commit it now.

**Comment by flawiddsouza** [comment]
Maybe it's some edge case. Oh well.

**Comment by peter-nl** [comment]
Hi all,
I could dive deeper into this but just to make sure I'm spending my time useful:
The current (master) gallery shortcode implementation is based on images uploaded (attached) to the page that holds the gallery shortcode. Whereas the gallery shortcode supports to use just any image, be it uploaded to this page or not. 
For the latter case variable 'ids' is used - which is not being processed in function bootstrap_wp_gallery, that only processes variable 'id'. Thus your miles may vary depending on what was uploaded to the page that hold the gallery shortcode.
So, the current function needs to be extended:
- if 'ids' is set, use those images
- otherwise use the images attached to the page that contains the shortcode
Ref function gallery_shortcode().
Does this make sense?

**Comment by holger1411** [comment]
For this release I removed the roots.io gallery solution because wordpress.org don´t accept add_shortcode:
WARNING: The theme uses the add_shortcode() function. Custom post-content shortcodes are plugin-territory functionality.

I removed the function and add the basic scss styls from underscores into it. See:
https://github.com/holger1411/understrap/commit/e5e91cead6ad315de49dfbbb340e93deacd951d6

Its not the perfect way becaus we use two differend css grids in one file for the same purpose (one from underscores for the gallery grid and one from Bootstrap4 for the rest...)

IMO it would be the best to filter the existing WP gallery shortcode markup to inject the BS4 classes. Will create an issue for that.
